### ISSUE TITLE
Switch from airbnb config to vercel?

### ISSUE THREAD

**Issue by frytg** [swrlab/eslint-plugin-swr#137]
We could evaluate whether switching to the Vercel style guide (or others) would be good for us, instead of this package or just the airbnb config
https://github.com/vercel/style-guide

**Comment by weiland** [comment]
I will collect some information on different styleguides / configs.

**Comment by weiland** [comment]
# Styleguide / Configs

## Requirements (to be defined properly below)

- drop-in / zero-config
- easy to use
- modular
- JS / (TS)
- basic code style
- fast
- IDE support
- cli / ci support
- extensive

## Current state

- airbnb styleguide

### Cons / Disadvantages

- airbnb is not well maintained
- too strict peer dependencies 
- newer approaches
- no typescript support


## Other approaches

### Vercel

#### Pros
- all configs (ts, eslint, prettier) are at one place
- good modularity 

#### Cons
- heavily relies on react 
	- but we could just skip these parts and integrate rules for vue

### unjs / nuxt

#### Pros
- comes from vue universe

#### Cons
- separate prettier and eslint

### Google

https://github.com/google/eslint-config-google

- archived and deprecated -> newer projects seem to rely more on TypeScript tooling


## Next moves

### Identify all requirements

- [x] Will many external (perhaps Rundfunk or agency but non audio-lab folks) be involved in using these configs?
- [x] Shall we introduce support for TS? (not yet)
- [x] Shall eslint embed the prettier config? -> separate (but mandatory rules are reflected in eslint too)
- Pro: eslint would print prettier errors in IDE 
- Cons: coupling, prettier should work independently 
- [x] use ESM? (is it even possible entirely?) (not yet, and only prettier first, eslint with flat config later and when repos are ESM too)
		
### Open questions
		
- [x] Create separate style-guide repo (and npm package) containing all configs (and archive old eslint and prettier) or archive prettier and re-work eslint repo
	- I would favour the first approach. 
	- -> start with a fork

Personally, I think that creating a style-guide-like repo would be a good approach to keep all configs in one place

**Comment by weiland** [comment]
Premises:

- Every project contains a `.editorconfig` file with the most important settings (tabs, `lf`, trailing lines in markdown etc).
- Every JS project contains additionally a *prettier* and *eslint* config.
- Atm there are no TypeScript projects.


- [x] create `@swrlab/@style-guide` repo (forked vercel for now)
- [x] import our previous rules (prettier & eslint)
- [x] fix `print_width` to `max_line_length` in *.editorconfig* so Prettier uses it (if there is no `printWidth` value set)
- [x] remove `printWidth` so the value from *.editcrconfg* is used 
- [x] create temporary `eslint` *migration* preset (which contains old settings for now)
- [x] port editorconfig/prettier config to *Biome*.json configuration (for experimental Biome support)
  There are only minor differences to prettier. 

Changes (for devs):

- Prettier config is separated to eslint (both run independenlty)
- There is no `tabWidth` value in the *.editorconfig* file -> set this value in your local editor.

Projects-Migration:

- [ ] Make sure that all `print_width` values in *.editorconfig* are replaced with `max_line_length`.
- [ ] Migrate prettier- and eslint-configs to new `style-guide`  
- [  ] and then fix possible issues.
- [ ] When there were many changes, create a `.git-rev` file with the *commit-Id* of the migration/formatting commit: https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt This helps *git* to ignore commit(s) in *git-blame*.
- [ ] Remove old packages from repo.

There is an example for the dashboard: 
- with prettier: https://github.com/swrlab/swr-lab-dashboard/tree/chore/switch-to-style-guide-with-all-pretter
- without prettier

Next steps/todos:

- [ ] Evaluate prettier vs [_@stylistic_](https://eslint.style/) via https://antfu.me/posts/why-not-prettier
  TLDR: eslint stylistic rules can do the same as prettier and there are way more configuration options + they don't enforce *printWidth* and don't break all long lines (when they sometimes make more sense). On the other hand: Prettier makes all the code look the same everywhere.
- [ ] Evaluate new *style-guide* prettier config
- [ ] Evaluate new *style-guide* eslint rules 
  - [ ] `unicorn/prefer-node-protocol` We should definitely do it.
  - [ ] `eslint-comments/require-description` would require many comments (on *directives*)
  - [ ] `unicorn/filename-case` we would need to change quite a few filenames (but seems alright)
  - [ ] `import/order` seems like a good rule
  - [ ] `import/no-default-export` (no strong opinion about but; we have many default exports)
  - [ ] `no-implicit-coercion` seems alright

**Comment by frytg** [comment]
testing new config with https://github.com/swrlab/style-guide
